### PR TITLE
Make composable to take pictures of parks

### PR DESCRIPTION
**Pull request by Alex720p**
This pr aims to allow users to take pictures and allowing us to process them. It introduces 2 new composables: `AddImageButton` and `ConfirmImageDialog`. The content is purely UI related.

**Summary:**
The `AddImageButton` composable handles image capture. It first checks and requests camera permission if needed. When permission is granted, it launches the camera to take a picture and stores it in a temporary file (in our app's cache). After the picture is taken, a `ConfirmImageDialog`is displayed to allow the user to either upload the image or cancel the operation. On confirmation or cancellation, the temporary file is cleaned up for privacy.

**Changes in existing composables:**
- `ParkOverview.kt`
  - Display the `AddImageButton` in `ImageTitle` and changed the title param to currentPark to allow us to link the images to the park later on.

Here's how it looks (`AddImageButton` is displayed on the top right of the park image):

<div style="display: flex; gap: 10px;"> Screenshot 1 Screenshot 2 Screenshot 3 Screenshot 4 </div>

Note: I'm pretty sure the dialog will break if we turn the screen to horizontal but since the ParkOverview screen already breaks, I won't bother implementing it now. The dialog will rescale correctly depending on phone sizes though.

### CODE REVIEW

File: app/src/main/java/com/android/streetworkapp/ui/image/AddImageButton.kt [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  const val DEBUG_PREFIX = "AddImageButton:"
+}
+
+/**
+ * IconButton that on onclick will request permission (if not already done) and open the camera to
+ * take a picture. After a picture is taken, a [ConfirmImageDialog] will be shown to prompt the user
+ * to cancel or upload the file
+ *
+ * @param currentPark The info of the park the image will be linked to
+ * @param context The current context
```
Comment (alvaro1080) [inline app/src/main/java/com/android/streetworkapp/ui/image/AddImageButton.kt:46]: Docs: the "context" is said to be a parameter in the docs, but it is not one of the parameters of the function

File: app/src/main/java/com/android/streetworkapp/ui/image/AddImageButton.kt [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+ */
+@Composable
+fun AddImageButton(currentPark: Park?) {
```
Comment (alvaro1080) [inline app/src/main/java/com/android/streetworkapp/ui/image/AddImageButton.kt:49]: Question: will the park parameter be used in the future? (if I am not mistaken, it is at the moment not used in the given function)

File: app/src/main/java/com/android/streetworkapp/ui/image/AddImageButton.kt [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  }
+
+  var showConfirmationDialog by remember { mutableStateOf(false) }
+  var capturedImageUri by remember { mutableStateOf<Uri?>(null) }
+
+  // Temporary file for storing the captured image
+  val tempFile by remember {
+    mutableStateOf(
+        File(context.cacheDir, "temp_image_for_park_upload.jpg").apply {
+          if (exists()) {
```
Comment (alvaro1080) [inline app/src/main/java/com/android/streetworkapp/ui/image/AddImageButton.kt:66]: Sugestion: those "if" perhaps be used in the same line , using a "&&" operator:

if(exists() && !delete()) 

(I might be missing something, if there is a reason for this nested if, perhaps we could add a small comment in the code)

File: app/src/main/java/com/android/streetworkapp/ui/park/ParkOverview.kt [hunk 4]
```diff
@@ -31,9 +31,11 @@ import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.MutableState
 import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableIntStateOf
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
```
Comment (alvaro1080) [inline app/src/main/java/com/android/streetworkapp/ui/park/ParkOverview.kt:38]: Suggestion: I think that this import can be removed as it is unused.

File: app/src/main/java/com/android/streetworkapp/ui/park/ParkOverview.kt [hunk 5]
```diff
@@ -31,9 +31,11 @@ import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.MutableState
 import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
```
Comment (alvaro1080) [inline app/src/main/java/com/android/streetworkapp/ui/park/ParkOverview.kt:34]: Suggestion: I think that this import can be removed as it is unused.

**Review by alvaro1080 — CHANGES_REQUESTED** [review]
Thank you for this Pull Request! It will be really useful to make sure that all parks have a unique picture, which is really great in terms of user experience!

I currently do not have a real device that can test this feature, I think that a second review with someone who could test the feature in real life would be beneficial.

I have a few comments and suggestions that I would like to discuss:

File: app/src/main/java/com/android/streetworkapp/ui/image/AddImageButton.kt (hunk 1 above)
Comment (Alex720p) [reply app/src/main/java/com/android/streetworkapp/ui/image/AddImageButton.kt:46]: my bad, made some last minute changes and forgot to remov it

File: app/src/main/java/com/android/streetworkapp/ui/image/AddImageButton.kt (hunk 2 above)
Comment (Alex720p) [reply app/src/main/java/com/android/streetworkapp/ui/image/AddImageButton.kt:49]: yep, when uploading an image, the collection of images will be linked by it's uid in Park (as progression is in user)

File: app/src/main/java/com/android/streetworkapp/ui/image/AddImageButton.kt (hunk 3 above)
Comment (Alex720p) [reply app/src/main/java/com/android/streetworkapp/ui/image/AddImageButton.kt:66]: done

File: app/src/main/java/com/android/streetworkapp/ui/park/ParkOverview.kt (hunk 4 above)
Comment (Alex720p) [reply app/src/main/java/com/android/streetworkapp/ui/park/ParkOverview.kt:38]: done

**Pull request merged**